test/e2e/scrapeconfig_test.go

```diff
@@ -3197,3 +3200,549 @@ var OpenStackSDTestCases = []scrapeCRDTestCase{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					NameFilter: ptr.To("my-server"),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid NameFilter",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					NameFilter: ptr.To(""),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing NameFilter",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Valid TagsFilter",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					TagsFilter: []string{"do"},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Empty tag values in TagsFilter",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					TagsFilter: []string{""}, // empty tag
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Repeating tags in TagsFilter",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleInstance,
+					AccessKey: "AccessKey",
+					SecretKey: v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "secret",
+						},
+						Key: "key.pem",
+					},
+					TagsFilter: []string{"do", "do"}, // repeating tags
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing TagsFilter",
```

Comment:
Same, and so on for all the optional fields if missing, just the valid and invalid should work